pipeline_steps/common.groovy

```diff
@@ -937,8 +937,12 @@ void standard_job_slave(String slave_type, Closure body){
         body()
       }
     } else if (slave_type == "container"){
+      String image_name = env.BUILD_TAG.toLowerCase()
+      // NOTE(mattt): This is necessary to initially gate this change as we
+      // have no parameter for SLAVE_CONTAINER_BASE_IMAGE defined yet
+      String base_image = env.SLAVE_CONTAINER_BASE_IMAGE ? env.SLAVE_CONTAINER_BASE_IMAGE : "ubuntu:16.04"
       dir("rpc-gating"){
-        container = docker.build env.BUILD_TAG.toLowerCase()
+        container = docker.build(image_name, "--build-arg BASE_IMAGE=${base_image} -f Dockerfile.standard_job .")
```

Comment:
How about we just combine the two approaches and creating something a bit more general, for example if we define the following:
SLAVE_CONTAINER_DOCKERFILE_REPO: ["RE", "PROJECT"]
SLAVE_CONTAINER_DOCKERFILE_PATH: file path to dockerfile relative to repo
SLAVE_CONTAINER_DOCKERFILE_BUILD_ARGS

this code could look something like:
```

if (env.SLAVE_CONTAINER_DOCKERFILE_REPO == "RE") {
    dockerfile_repo_dir = "${env.WORKSPACE}/rpc-gating"
} elif (env.SLAVE_CONTAINER_DOCKERFILE_REPO == "PROJECT") {
    // If pr clone
        common.clone_with_pr_refs("${{env.WORKSPACE}}/${{env.RE_JOB_REPO_NAME}}")
    // else clone branch
        ...
    dockerfile_repo_dir = "${{env.WORKSPACE}}/${{env.RE_JOB_REPO_NAME}}"
} else {
    BOOM!
}
dir(dockerfile_repo_dir) {
    container = docker.build(image_name, "${SLAVE_CONTAINER_DOCKERFILE_BUILD_ARGS} -f ${SLAVE_CONTAINER_DOCKERFILE_PATH} .")
}
```

Reply:
This seems like an entirely reasonable suggestion to me and I can't think of any reason why this wouldn't work.  @hughsaunders this look good to you before I test this out?

Reply:
👍

Reply:
@git-harry is there any reason why we can't just merge `SLAVE_CONTAINER_DOCKERFILE_PATH` into `SLAVE_CONTAINER_DOCKERFILE_BUILD_ARGS`?  For example:

```
SLAVE_CONTAINER_DOCKERFILE_BUILD_ARGS="--build-arg BASE_IMAGE=ubuntu:17.10 -f Dockerfile"
```

Reply:
@mattt416 I guess it depends how much control we want to retain. In my mind `SLAVE_CONTAINER_DOCKERFILE_BUILD_ARGS` would actually be a list of args that are then used to build part of the command, something vaguely like:
```
build_args = ""
for arg in SLAVE_CONTAINER_DOCKERFILE_BUILD_ARGS:
    quoted_arg = \\ do groovy equivalent of https://docs.python.org/dev/library/shlex.html#shlex.quote
    build_args += " --build-arg ${quoted_arg}"
```

I'm not sure if there are any options that `docker build` takes that we would explicitly want to exclude but my natural inclination would be to manage what can be set to prevent the unexpected. If this can only be set in the project then maybe reviews of rpc-gating are a sufficiently strong control.